src/binder/assemblybinder.cpp

```diff
@@ -685,19 +685,23 @@ namespace BINDER_SPACE
 
         StackSString sCoreLib;
 
-        // At run-time, System.Private.CoreLib.ni.dll is typically always available, and
-        // System.Private.CoreLib.dll is typically not.  So check for the NI first.
+        // At run-time, System.Private.CoreLib.dll is expected to be the NI image.
+        // Try binding to it first as that. If it fails, we will try to bind to it as 
+        // IL image.
         sCoreLib = sCoreLibDir;
-        sCoreLib.Append(CoreLibName_NI_W);
+        sCoreLib.Append(CoreLibName_IL_W);
+#if defined(FEATURE_PAL) && !defined(_TARGET_AMD64_)
+        // Native image for core library only exists for X64 on non-Windows. As other architectures
```

Comment:
The only real difference that fExplicitBindToNativeImage makes is whether we throw when LoadLibrary fails, or whether we take the fallback path. It is done here: https://github.com/dotnet/coreclr/blob/3c61ffd676603bb14d4e815e1e249f816d4cbc74/src/vm/peimage.cpp#L918. I think we should be able to remove all this if we start unconditionally passing in `bThrowOnError = FALSE` here.